core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -210,21 +210,23 @@ class RangePolicy : public Impl::PolicyTraits<Properties...> {
  private:
   /** \brief finalize chunk_size if it was set to AUTO*/
   inline void set_auto_chunk_size() {
-    typename traits::index_type concurrency =
-        traits::execution_space::concurrency();
+    int64_t concurrency =
+        static_cast<int64_t>(traits::execution_space::concurrency());
     if (concurrency == 0) concurrency = 1;
 
     if (m_granularity > 0) {
       if (!Impl::is_integral_power_of_two(m_granularity))
         Kokkos::abort("RangePolicy blocking granularity must be power of two");
     }
 
-    member_type new_chunk_size = 1;
-    while (new_chunk_size * 100 * concurrency < m_end - m_begin)
+    int64_t new_chunk_size = 1;
+    while (new_chunk_size * 100 * concurrency <
```

Comment:
Is there any reason to change the type for `new_chunk_size`?
In the end, it needs to be assignable to `m_granularity` (which is of type `member_type`) anyway and the comparison should work as expected because `concurrency` is `int64_t`.

Reply:
Because new_chunk_size is overflowing ...